openssl/openssl_lock_setup.c

```diff
@@ -27,7 +26,7 @@ static MUTEX_TYPE *mutex_buf = NULL;
  
 static void locking_function(int mode, int n, const char *file, int line)
 {
-  if(mode & CRYPTO_LOCK)
+  if(mode & 1)
```

Comment:
Hmm, it seems to me that it might be worth keeping this name around, to keep it in context, since this is a flag constant, not just a boolean.

```
        CRYPTO_LOCK     0x01
        CRYPTO_UNLOCK   0x02
        CRYPTO_READ     0x04
        CRYPTO_WRITE    0x08
```